### DISCUSSION TITLE
IFitsHeader errors? Should MaxImDLExt.BAYERPAT and MaxImDLExt.PIERSIDE both be String type instead of real?

### DISCUSSION THREAD

**Discussion by johnmurphyastro** [nom-tam-fits/nom-tam-fits#655] [General]
Hi, thanks for this superb library, and all the work that has gone into it.
A possible minor correction:
Should the IFitsHeader values for
**MaxImDLExt.BAYERPAT**
**MaxImDLExt.PIERSIDE**
be VALUE.STRING instead of VALUE.REAL ?

Also, does a predefined entry for **ROWORDER** exist? I have created my own, but it would be great if this was included.
Regards, John Murphy

**Comment by attipaci** [comment]
Ho @johnmurphyastro,

That is quite possible indeed. The `MaxImDLExt` enums were added before my time, so I'm not intricately familiar with it. I just took a peek at the documentation at [https://cdn.diffractionlimited.com/help/maximdl/FITS_File_Header_Definitions.htm](https://cdn.diffractionlimited.com/help/maximdl/FITS_File_Header_Definitions.htm), and from there it's not at all clear what type of values many of these keywords should take, including the ones you mentioned.

Do you know of a more detailed documentation that more explicitly defines the type of values for these keywords? Or, perhaps some usage examples?

In any case, I'll gladly change the associated types if you are certain that those you mentioned are the definitive ones. Otherwise, we can mark them as `VALUE.ANY` to accept values of any type. Can you check that all other definitions are correct as well? -- so once we fix it, we won't have to come back to it again...

Thanks for noting the issue, and more thanks in advance for helping to get this corrected properly...

**Reply by johnmurphyastro** [reply]
Hi @attipaci

At the moment, all entries in MaxImDLExt are set to VALUE.REAL, which seems unlikely.

**I suggest the following MaxImDLExt corrections:**

// BAYERPAT examples: "RGGB", "GRBG", "GBRG", "BGGR", "RGBG", "GRGB", "GBGR", or "BGRG"
BAYERPAT(VALUE.STRING, "Image Bayer color pattern"),

// MaxIm DL Version 6.05 sets this to a blank string in my images. I've not seen it used else where.
FLIPSTAT(VALUE.ANY, "status of pier flip"),

// I've never seen this used, but VALUE.REAL seams unlikely.
INPUTFMT(VALUE.ANY, "format of file"),

// MaxIm DL Version 6.07 sets this to "EAST" or "WEST" in my images.
// Other software sets it to "East" or "West"
PIERSIDE(VALUE.STRING, "Side-of-pier status"),

// MaxIm DL Version 6.07 sets this to "Raw" in my images
READOUTM(VALUE.STRING, "Readout Mode for the camera"),

**The following would be great additions (frequently used with amateur cameras):**

ANGLE(VALUE.REAL, "Image angle"),

// "2017-01-03T02:41:24" or "2024-02-24T22:23:33.054"
DATE_LOC("DATE-LOC", VALUE.STRING, "Time of observation (local)"),

// Always true or false
FLIPPED(VALUE.LOGICAL, "Is image flipped"),

FOCRATIO(VALUE.REAL, "Focal ratio"),

// Very common since CMOS cameras became popular
OFFSET(VALUE.INTEGER, "Camera offset setting"),

// "TOP-DOWN" or "BOTTOM-UP"
ROWORDER(VALUE.STRING, "FITS image orientation"),    

// Synonyms of other entries, but frequently used
CCDXBIN(VALUE.INTEGER, "X axis binning factor"),
CCDYBIN(VALUE.INTEGER, "Y axis binning factor"),
FOCPOS(VALUE.REAL, "Focuser position in steps"),
FOCTEMP(VALUE.REAL, "Focuser temperature readout")

CLOUDCVR(VALUE.REAL, "[percent] Cloud cover"),
HUMIDITY(VALUE.REAL, "[percent] relative humidity"),
DEWPOINT(VALUE.REAL, "[degC] Dew point"),
AMBTEMP(VALUE.REAL, "[degC] Ambient air temperature"),
WINDSPD(VALUE.REAL, "[kph] Wind speed"),
WINDDIR(VALUE.REAL, "[deg] Wind direction: 0=N, 180=S, 90=E, 270=W"),
PRESSURE(VALUE.REAL, "[hPa] Air pressure")

Regards, John Murphy

**Comment by johnmurphyastro** [comment]
Thanks for making the changes.
